onboarding/source/amb_light_service.c

```diff
@@ -18,22 +27,99 @@ static void lightServiceTask(void * pvParameters);
 uint8_t initLightService(void) {
     /* USER CODE BEGIN */
     // Create the task and queue here.
+    BaseType_t xReturned = pdFAIL;
+    if(lightServiceTaskHandle == NULL){
+        xReturned = xTaskCreate(lightServiceTask,
+                                LIGHT_SERVICE_NAME,
+                                LIGHT_SERVICE_STACK_SIZE,
+                                NULL,
+                                LIGHT_SERVICE_PRIORITY,
+                                &lightServiceTaskHandle);
 
+    }
+
+    if(xReturned == pdFAIL){
+        return 0;
+    }
+
+    
+    
+    if(xLightServiceQueue == NULL){
+	    
+	xLightServiceQueue = xQueueCreate(LIGHT_SERVICE_QUEUE_SiZE, LIGHT_EVENT_SIZE);
+	    
+	if(xLightServiceQueue == NULL){
+        	return 0;
+	}
+    }
     /* USER CODE END */
     return 1;
 }
 
+uint16_t getLightSensorData(void)
+{
+
+	adcData_t lightData;
+    	lightData.id = 6;
+	adcData_t *adcDataPtr = &lightData;
+
+ 	/** - Start Group1 ADC Conversion 
+ 	*     Select Channel 6 - Light Sensor for Conversion
+ 	*/
+	adcStartConversion(adcREG1, adcGROUP1);
+
+ 	/** - Wait for ADC Group1 conversion to complete */
+ 	while(!adcIsConversionComplete(adcREG1, adcGROUP1)); 
+
+	/** - Read the conversion result
+	*     The data contains the Light sensor data
+    */
+	adcGetData(adcREG1, adcGROUP1, adcDataPtr);
+	
+	/** - Transmit the Conversion data to PC using SCI
+    */
+	return (adcDataPtr->value);
+}
+
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
+    ASSERT(lightServiceTaskHandle != NULL);
+    light_event_t queueTask = NULL_EVENT;
+	
+        if (xQueueReceive(xLightServiceQueue, &queueTask, portMAX_DELAY) == pdTRUE) {
+		
+            if (queueTask == MEASURE_LIGHT){
+            
+            	uint16_t lightSensorData = getLightSensorData();
+            	char data[sizeof(lightSensorData)];
+            	int length = snprintf(data, sizeof(data), "%c", lightSensorData);
 
+            	sciPrintText(scilinREG, (unsigned char*)data, length + 1);
```

Comment:
This can lead to u accessing memory u haven't allocated. `snprintf` returns:
```
The number of characters that would have been written if n had been sufficiently large, not counting the terminating null character.
If an encoding error occurs, a negative number is returned.
Notice that only when this returned value is non-negative and less than n, the string has been completely written.
```
`length + 1` can be bigger than the size of your buffer. You don't want that.